tests/scripts/mocksocktests.sh

```diff
@@ -0,0 +1,50 @@
+#!/bin/sh
+
+# Usage: ./mocksocktests.sh /path/to/clusterclient-binary
+
+clientprog=${1:-./clusterclient}
+
+# == Expected traffic ==
+# Client: ["CLUSTER", "SLOTS"]
+# Server: [[0, 16383, ["127.0.0.1", 30001, "nodeid123"]]]
+# Client: (reconnect)
+# Client: SET foo bar
+# Server: +OK
+# Client: GET foo
+# Server: bar
+# Client: (close)
+printf '%s\n' \
+       'RECV *2\r\n$7\r\nCLUSTER\r\n$5\r\nSLOTS\r\n' \
+       'SEND *1\r\n*3\r\n:0\r\n:16383\r\n*3\r\n$9\r\n127.0.0.1\r\n:30001\r\n$9\r\nnodeid123\r\n' \
+       'PEER RECONNECT' \
+       'RECV *3\r\n$3\r\nSET\r\n$3\r\nfoo\r\n$3\r\nbar\r\n' \
+       'SEND +OK\r\n' \
+       'RECV *2\r\n$3\r\nGET\r\n$3\r\nfoo\r\n' \
+       'SEND $3\r\nbar\r\n' \
+       'PEER CLOSE' \
+    | timeout 5s ./mocksock.pl -p 30001 -d &
```

Comment:
the timeout 5s, that will kill ./mocksock.pl after 5s?

Reply:
Yes, well, it will send a SIGTERM, which makes it close the listener socket and then exit with non-zero status.